openmetadata-ui/src/main/resources/ui/src/components/common/GlossaryTermSelectableList/GlossaryTermSelectableList.component.tsx

```diff
@@ -0,0 +1,156 @@
+ *  limitations under the License.
+ */
+import { Popover, Space, Typography } from 'antd';
+import { useState } from 'react';
+import { useTranslation } from 'react-i18next';
+import { ReactComponent as IconTerm } from '../../../assets/svg/book.svg';
+import {
+  ADD_USER_CONTAINER_HEIGHT,
+  DE_ACTIVE_COLOR,
+} from '../../../constants/constants';
+import { EntityReference } from '../../../generated/entity/data/table';
+import {
+  LabelType,
+  State,
+  TagLabel,
+  TagSource,
+} from '../../../generated/type/tagLabel';
+import { fetchGlossaryList } from '../../../utils/TagsUtils';
+import { FocusTrapWithContainer } from '../FocusTrap/FocusTrapWithContainer';
+import { SelectableList } from '../SelectableList/SelectableList.component';
+import { GlossaryTermSelectableListProps } from './GlossaryTermSelectableList.interface';
+
+export const GlossaryTermListItemRenderer = (props: EntityReference) => {
+  return (
+    <Space>
+      <IconTerm
+        className="align-middle"
+        color={DE_ACTIVE_COLOR}
+        height={16}
+        name="doc"
+        width={16}
+      />
+      <Typography.Text>{props.displayName || props.name}</Typography.Text>
+    </Space>
+  );
+};
+
+export const GlossaryTermSelectableList = ({
+  selectedTerms = [],
+  onUpdate,
+  children,
+  popoverProps,
+  onCancel,
+  listHeight = ADD_USER_CONTAINER_HEIGHT,
+}: GlossaryTermSelectableListProps & { listHeight?: number }) => {
+  const [popupVisible, setPopupVisible] = useState(false);
+  const { t } = useTranslation();
+
+  const convertTermsToEntityReferences = (
+    terms: TagLabel[]
+  ): EntityReference[] => {
+    return terms.map((term) => ({
+      id: term.tagFQN || '',
+      name: term.name || term.tagFQN || '',
+      displayName: term.displayName || term.name || term.tagFQN,
+      type: 'glossaryTerm',
+      fullyQualifiedName: term.tagFQN,
+      description: term.description,
+    }));
+  };
+
+  const convertEntityReferencesToTerms = (
+    refs: EntityReference[]
+  ): TagLabel[] => {
+    return refs.map((ref) => ({
+      tagFQN: ref.fullyQualifiedName || ref.id,
+      displayName: ref.displayName || ref.name,
+      name: ref.name,
+      source: TagSource.Glossary,
+      labelType: LabelType.Manual,
+      state: State.Confirmed,
+      description: ref.description,
+    }));
+  };
+
+  const fetchGlossaryTermOptions = async (
+    searchText: string,
+    after?: string
+  ) => {
+    try {
+      const afterPage = after ? parseInt(after, 10) : 1;
+      const response = await fetchGlossaryList(searchText, afterPage);
+      const terms = response.data || [];
+
+      const entityRefs: EntityReference[] = terms.map((term) => ({
+        id: term.value,
+        name: term.data?.name || term.label,
+        displayName: term.data?.displayName || term.label,
+        type: 'glossaryTerm',
+        fullyQualifiedName: term.value,
+        description: term.data?.description,
+      }));
```

Comment:
Can we avoid conversion on both side or it's needed

Reply:
The conversions are necessary because they bridge two incompatible data structures:
API returns glossary term format but SelectableList component expects EntityReference[] format and parent callback needs original term format
Cannot avoid both-way conversion, but we optimized with useCallback to prevent function recreation on every render.